Storage/SQL/SQLiteHistory.swift

```diff
@@ -320,24 +325,28 @@ extension SQLiteHistory: BrowserHistory {
         }
 
         let ungroupedSQL =
-        "SELECT \(TableHistory).id AS historyID, \(TableHistory).url AS url, title, guid, domain_id, domain, " +
-        "COALESCE(max(case \(TableVisits).is_local when 1 then \(TableVisits).date else 0 end), 0) AS localVisitDate, " +
-        "COALESCE(max(case \(TableVisits).is_local when 0 then \(TableVisits).date else 0 end), 0) AS remoteVisitDate, " +
-        "COALESCE(count(\(TableVisits).is_local), 0) AS visitCount " +
-        "FROM \(TableHistory) " +
-        "INNER JOIN \(TableDomains) ON \(TableDomains).id = \(TableHistory).domain_id " +
-        "INNER JOIN \(TableVisits) ON \(TableVisits).siteID = \(TableHistory).id " +
-        whereClause + " GROUP BY historyID"
+            "SELECT \(TableHistory).id AS historyID, \(TableHistory).url AS url, title, guid, domain_id, domain, " +
+            "COALESCE(max(case \(TableVisits).is_local when 1 then \(TableVisits).date else 0 end), 0) AS localVisitDate, " +
+            "COALESCE(max(case \(TableVisits).is_local when 0 then \(TableVisits).date else 0 end), 0) AS remoteVisitDate, " +
+            "COALESCE(count(\(TableVisits).is_local), 0) AS visitCount " +
+            (includeBookmarked ? ", count(\(TableBookmarks).id) AS bookmarked " : "") +
+            "FROM \(TableHistory) " +
+            (includeBookmarked ? "LEFT OUTER JOIN \(TableBookmarks) ON \(TableBookmarks).url IS \(TableHistory).url " : "") +
```

Comment:
I have a gentle preference for keeping the zero indent to buy more column width for SQL, and a moderate preference for not breaking blame — particularly not mixed in with a patch that makes other changes. If you're going to change spacing on lots of lines, do it in a separate commit so the real changes are easier to see.